addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.js

```diff
@@ -143,10 +195,79 @@ class ComposerTextInput extends Component {
         this._textareaRef.el.style.height = (this._textareaRef.el.scrollHeight) + "px";
     }
 
+    /**
+     * Validates a keyword in order to trigger mentions search.
+     *
+     * @private
+     * @param delimiter
+     * @param beginningOnly
+     * @return {*}
+     */
+    _validateKeyword(delimiter, beginningOnly) {
+        const leftString = this.getContent().substring(0, this._getSelectionStart());
+
+        // use position before delimiter because there should be whitespaces
+        // or line feed/carriage return before the delimiter
+        const beforeDelimiterPosition = leftString.lastIndexOf(delimiter) - 1;
+        if (beginningOnly && beforeDelimiterPosition > 0) {
+            return false;
+        }
+        let searchStr = this.getContent().substring(beforeDelimiterPosition, this._getSelectionStart());
+        // regex string start with delimiter or whitespace then delimiter
+        const pattern = "^"+delimiter+"|^\\s"+delimiter;
+        const regexStart = new RegExp(pattern, 'g');
+        // trim any left whitespaces or the left line feed/ carriage return
+        // at the beginning of the string
+        searchStr = searchStr.replace(/^\s\s*|^[\n\r]/g, '');
+        if (regexStart.test(searchStr) && searchStr.length) {
+            searchStr = searchStr.replace(pattern, '');
+            return searchStr.indexOf(' ') < 0 && !/[\r\n]/.test(searchStr)
+                ? searchStr.replace(delimiter, '')
+                : false;
+        }
+        return false;
+    }
+
+    /**
+     * Detects if mentioned partners are still in the composer text input
+     * and removes them if not.
+     *
+     * @private
+     */
+    _validatePartnerIds() {
+        const inputMentions = this.getContent().match(new RegExp("@"+'[^ ]+(?= |&nbsp;|$)', 'g'));
+        if (inputMentions) {
+            for (const id in this.composer.mentionedPartners) {
+                const mention = this.composer.mentionedPartners[id];
+                let inputMention = inputMentions.find(item => {
+                    return item === ("@" + mention.name).replace(/ /g, '\u00a0');
+                });
+                if (!inputMention) {
+                    this.composer.mentionedPartners.splice(this.composer.mentionedPartners.indexOf(mention), 1);
+                }
+            }
+        }
+        else {
+            this.composer.unlink({ mentionedPartners: null });
+        }
+    }
+
     //--------------------------------------------------------------------------
     // Handlers
     //--------------------------------------------------------------------------
 
+    /**
+     * @private
+     * @param ev
+     */
+    _onClickMentionItem(ev) {
+        ev.preventDefault();
+        this.composer._insertMentionSuggestion(this.composer.active_id, this._mentionWord);
+        this._update();
```

Comment:
no, let re-render cycle do the `_update`

Reply:
This should be possible to be removed, you can try again now that other changes have been done, and if still not possible, further investigation is needed. Maybe just some keys are missing in useStore?